climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _comparison_type: str
+    _hindcast_comparison: bool
+    _probabilistic_comparison: bool
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+
+    def factory(self, type) -> 'Comparison':
+        try:
+            return comparison_dict[type](self._prediction, self._observation)
```

Comment:
I think type is a reserved word; use `type_`, if not `method` or `kind`
comparison_type
comparison_method
comparison_kind